From 15fcdeca3e62f8093c19d484c5780f7a89205b27 Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Wed, 24 Jan 2024 18:04:53 -0500
Subject: [PATCH 092/105] Allow for adjusting cpu energy performance

Current value of the energy vs performance hint for the given policy (or the CPU represented by it).

Normal options: default, performance, balance_performance, balance_power, power

CPU_ENERGY_PERFORMANCE_PREF=balance_power
---
 init.sh | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/init.sh b/init.sh
index 38faf7e..a555937 100644
--- a/init.sh
+++ b/init.sh
@@ -39,6 +39,21 @@ function init_misc()
 		echo $INTEL_PSTATE_CPU_MAX_PERF_PCT > /sys/devices/system/cpu/intel_pstate/max_perf_pct
 	fi
 
+	# Allow for adjusting cpu energy performance
+	# Normal options: default, performance, balance_performance, balance_power, power
+	if [ -n "$CPU_ENERGY_PERFORMANCE_PREF"  ]; then
+
+		cpuprefavailable=$(cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences)
+		
+		# if the word "CPU_ENERGY_PERFORMANCE_PREF" is found in the $cpuprefavailable string, then set the options for all cpus
+		if [[ "$cpuprefavailable" == *"$CPU_ENERGY_PERFORMANCE_PREF"* ]]; then
+			for cpu in $(ls -d /sys/devices/system/cpu/cpu?); do
+				echo $CPU_ENERGY_PERFORMANCE_PREF > $cpu/cpufreq/energy_performance_preference || return 1
+			done
+		fi
+
+	fi
+
 	# enable sdcardfs if /data is not mounted on tmpfs or 9p
 	#mount | grep /data\ | grep -qE 'tmpfs|9p'
 	#[ $? -eq 0 ] && set_prop_if_empty ro.sys.sdcardfs false
-- 
2.34.1

